packages/native/src/useBackButton.tsx

```diff
@@ -16,7 +16,9 @@ export default function useBackButton(
         }
 
         if (navigation.canGoBack()) {
-          navigation.goBack();
+          navigation.dispatch({ type: 'CLOSE_DRAWER' }, () =>
+            setImmediate(navigation.goBack)
```

Comment:
I don't think it's desirable to both close the drawer and go back on back button press.

If there's such a use case, we can add an option in drawer navigator and internally it can listen to back button press and dismiss the drawer without. But I'm not convinced about this.